### ISSUE TITLE
Opening but not completing a codebase import or creation disrupts page displays

### ISSUE THREAD

**Issue by aptmac** [openshiftio/openshift.io#2732]
Description: Opening and not completing a codebase addition/import can result in unintended page layouts (overlapping components and black screens).

Gif:
![pr 2732 gif](https://user-images.githubusercontent.com/10425301/37791666-3192bed0-2de0-11e8-83e2-2dfcdc5900eb.gif)

Steps to reproduce:
1. Go to a space, and click the "Add Codebase" button to open the wizard. Exit the wizard (by closing the x, the no thanks button, or clicking off of the modal)
![2018-03-22-142545_1916x1033_scrot](https://user-images.githubusercontent.com/10425301/37790450-f31713fc-2ddc-11e8-9d6f-7cb2a1eaeff7.png)

2. Go to your profile page, notice that the spaces widget now shows the "Create Space" screen and a list of the user's spaces overlapped on top of each other.
![profile](https://user-images.githubusercontent.com/10425301/37790332-9441f57c-2ddc-11e8-805c-c228863e848e.png)

3. This is the trace when running locally, running in production doesn't seem to print the errors. The trace says the error is here: https://github.com/fabric8-ui/fabric8-ui/blob/0db1b8c07de4bbc667826f2c6c309e1ca243e0d9/src/app/space/analyze/analyze-overview/analyze-overview.component.html#L37 [0], and it doesn't like the [spaces]="space.attributes.name" because it's null.
![stack-trace](https://user-images.githubusercontent.com/10425301/37790333-94564e14-2ddc-11e8-9f08-edcf0d0e3486.png)

4. If you go back to your space and try to "Add Codebase" again, the page goes completely black.
![blank screen](https://user-images.githubusercontent.com/10425301/37790453-f60f7cac-2ddc-11e8-9a2d-c7018f17d52d.png)

[0] https://github.com/fabric8-ui/fabric8-ui/blob/0db1b8c07de4bbc667826f2c6c309e1ca243e0d9/src/app/space/analyze/analyze-overview/analyze-overview.component.html#L37

**Comment by aptmac** [comment]
Looking into this a bit deeper, there's a couple of things that may play a role in this:

1. Closing the codebases wizard doesn't run ngOnDestroy() for the component, not sure if that's intentional

2. When running a local f8ui with breakpoints inserted, I've noticed that the error message I saw about the analyze-overview component (step 3 image in original post) is getting hit on every page I view. In fact, it's ngOnInit() [1] is getting called every time I visit a new page. 

Once this state is triggered, visiting other parts of openshift.io will cause multiple page elements to be combined into the same page, in a similar manner to https://github.com/openshiftio/openshift.io/issues/1975 [2]
![broken](https://user-images.githubusercontent.com/10425301/37833856-d16042e6-2e82-11e8-8378-05e666a397df.png)


[1] https://github.com/fabric8-ui/fabric8-ui/blob/0db1b8c07de4bbc667826f2c6c309e1ca243e0d9/src/app/space/analyze/analyze-overview/analyze-overview.component.ts#L34
[2] https://github.com/openshiftio/openshift.io/issues/1975
